File: Python 3/Problem_025.py
# What is the first term in the Fibonacci sequence to contain 1000 digits?
from euler import fibonacci


def first_Fib_N_digits(N):
    for i, f in enumerate(fibonacci()):
        if f >= 10 ** (N - 1):
            return i + 1


if __name__ == "__main__":
    print(first_Fib_N_digits(1000))

File: Haskell/Problem_025.hs
-- What is the first term in the Fibonacci sequence to contain 1000 digits?
import           System.IO
import           Euler                          ( fibonacci )


firstFibNDigits :: Integer -> Integer
firstFibNDigits n = fst . head . dropWhile checkDigits $ zip [0 ..] fibonacci
  where checkDigits = (< 10 ^ (n - 1)) . snd


main :: IO ()
main = do
  print $ firstFibNDigits 1000
